src/foam/dao/ManyToManyRelationshipDAO.js

```diff
@@ -18,51 +18,38 @@
 foam.CLASS({
   package: 'foam.dao',
   name: 'ManyToManyRelationshipDAO',
-  extends: 'foam.dao.RelationshipDAO',
+  extends: 'foam.dao.ProxyDAO',
```

Comment:
This drops storing a predicate (and superclass implementation to filter by the predicate). Does that potentially eliminate the possibility of the `find()` implementation I propose below?